content/300-guides/050-database/100-developing-with-prisma-migrate/175-squashing-migrations.mdx

```diff
@@ -0,0 +1,124 @@
+
+## Considerations when squashing migrations
+
+<Admonition type="warning">
+
+When squashing migrations, be aware that any custom SQL in your `migration.sql` files will not be retained.
+
+</Admonition>
+
+## How to squash migrations
+
+This section provides step-by-step instructions on how to squash migrations in the two scenarios discussed above:
+
+- [Migrating cleanly from a development environment](#how-to-migrate-cleanly-from-a-development-environment)
+- [Creating a clean history in a production environment](#how-to-create-a-clean-history-in-a-production-environment)
+
+### How to migrate cleanly from a development environment
+
+Before squashing your migrations, make sure you have the following starting conditions:
+
+- The contents of the migrations to be squashed are not yet applied on the production database
+- All migrations applied to production are part of the local migration history already
+- There is no custom SQL in the migration files
+
+Then follow these steps:
+
+1. Reset the contents of your local `./migrations` folder to match the migration history on the `main` branch
+
+2. Create an empty `migration.sql` file in the `./migrations` folder and make a note of the file path
+
+3. Create a single migration that takes you:
+
+   - from the state of the `main` branch as described in your reset migration history
+   - to the state of your local feature as described in your `schema.prisma` file
+   - and outputs this to the `migration.sql` file created above
+
+   You can do this using the `migrate diff` command:
+
+   ```terminal
+   npx prisma migrate diff \
+   --preview-feature \
+   --from-migrations ./migrations \
+   --to-schema-datamodel ./schema.prisma \
+   --script > filepath
+   ```
+
+You should now have a single migration file that can be applied to production using `migrate deploy`.
+
+### How to create a clean history in a production environment
+
+Before squashing your migrations, make sure you have the following starting conditions:
+
+- All migrations in the migration history are applied on the production database
+- The datamodel matches the migration history
+- There is no custom SQL in the migration files
+
+Then follow these steps:
+
+1. Delete all contents of the `./migrations` folder
+
+2. Create an empty `migration.sql` file in the `./migrations` folder and make a note of the file path
+
+3. Create a single migration that takes you:
+
+   - from an empty database
+   - to the current state of the production database schema as described in your `schema.prisma` file
+   - and outputs this to the `migration.sql` file created above
+
+   You can do this using the `migrate diff` command:
+
+   ```terminal
+   npx prisma migrate diff \
+   --preview-feature \
+   --from-migrations ./migrations \
```

Comment:
using --from-empty here instead of --from-migrations should have the same effect, but maybe it is easier to understand?